Espresso Test failing in TopicPracticeFragmentTest

`testTopicPracticeFragment_loadFragment_selectSubtopics_clickStartButton_skillListTransferSuccessfully` is failing

<img width="1510" alt="Screenshot 2021-07-06 at 1 08 25 AM" src="https://user-images.githubusercontent.com/9396084/124512747-c151f800-ddf6-11eb-812e-2cebd0f17dc5.png">

The actual error here is:

```
Caused by: androidx.test.espresso.IdlingResourceTimeoutException: Wait for [TestCoroutineDispatcherIdlingResource] to become idle timed out
at androidx.test.espresso.IdlingPolicy.handleTimeout(IdlingPolicy.java:61)
```

If you see in the emulator when running the test, it did all the steps, the `QuestionPlayerActivity` is also opening, but for idling resource, before the execution comes to intended and check the activity, the time out thing occurs and it throws `THROW_IDLE_TIMEOUT`.

--- a/app/src/sharedTest/java/org/oppia/android/app/topic/practice/TopicPracticeFragmentTest.kt
+++ b/app/src/sharedTest/java/org/oppia/android/app/topic/practice/TopicPracticeFragmentTest.kt
@@ -4,6 +4,7 @@ import android.app.Application
 import androidx.appcompat.app.AppCompatActivity
 import androidx.recyclerview.widget.RecyclerView
 import androidx.test.core.app.ActivityScenario
+import androidx.test.core.app.ActivityScenario.launch
 import androidx.test.core.app.ApplicationProvider
 import androidx.test.espresso.Espresso.onView
 import androidx.test.espresso.action.ViewActions.click
@@ -91,6 +92,7 @@ import org.robolectric.annotation.Config
 import org.robolectric.annotation.LooperMode
 import javax.inject.Inject
 import javax.inject.Singleton
+import org.oppia.android.app.devoptions.DeveloperOptionsActivity
 
 /** Tests for [TopicPracticeFragment]. */
 @RunWith(AndroidJUnit4::class)
@@ -244,14 +246,49 @@ class TopicPracticeFragmentTest {
 
   @Test
   fun testTopicPracticeFragment_loadFragment_selectSubtopics_clickStartButton_skillListTransferSuccessfully() { // ktlint-disable max-line-length
-    testCoroutineDispatchers.unregisterIdlingResource()
-    launchTopicActivityIntent(internalProfileId, FRACTIONS_TOPIC_ID)
-    clickPracticeTab()
-    clickPracticeItem(position = 1, targetViewId = R.id.subtopic_check_box)
-    scrollToPosition(position = 5)
-    clickPracticeItem(position = 5, targetViewId = R.id.topic_practice_start_button)
-    intended(hasComponent(QuestionPlayerActivity::class.java.name))
-    intended(hasExtra(QuestionPlayerActivity.getIntentKey(), skillIdList))
+//    testCoroutineDispatchers.unregisterIdlingResource()
+    launchTopicActivityIntent(internalProfileId, FRACTIONS_TOPIC_ID).use {
+//      testCoroutineDispatchers.runCurrent()
+      onView(
+        allOf(
+          withText(TopicTab.getTabForPosition(position = 2, enablePracticeTab).name),
+          isDescendantOfA(withId(R.id.topic_tabs_container))
+        )
+      ).perform(click())
+      onView(
+      atPositionOnView(
+        recyclerViewId = R.id.topic_practice_skill_list,
+        position = 1,
+        targetViewId = R.id.subtopic_check_box
+      )
+    ).perform(click())
+      onView(withId(R.id.topic_practice_skill_list)).perform(
+        scrollToPosition<RecyclerView.ViewHolder>(
+          5
+        )
+      )
+      onView(
+      atPositionOnView(
+        recyclerViewId = R.id.topic_practice_skill_list,
+        position = 5,
+        targetViewId = R.id.topic_practice_start_button
+      )
+    ).perform(click())
+      testCoroutineDispatchers.runCurrent()
+      intended(hasComponent(QuestionPlayerActivity::class.java.name))
+      intended(hasExtra(QuestionPlayerActivity.getIntentKey(), skillIdList))
+
+      /*clickPracticeTab()
+      testCoroutineDispatchers.runCurrent()
+      clickPracticeItem(position = 1, targetViewId = R.id.subtopic_check_box)
+      testCoroutineDispatchers.runCurrent()
+      scrollToPosition(position = 5)
+      testCoroutineDispatchers.runCurrent()
+      clickPracticeItem(position = 5, targetViewId = R.id.topic_practice_start_button)
+      testCoroutineDispatchers.runCurrent()
+      intended(hasComponent(QuestionPlayerActivity::class.java.name))
+      intended(hasExtra(QuestionPlayerActivity.getIntentKey(), skillIdList))*/
+    }
   }
 
   @Test

--- a/app/src/sharedTest/java/org/oppia/android/app/topic/practice/TopicPracticeFragmentTest.kt
+++ b/app/src/sharedTest/java/org/oppia/android/app/topic/practice/TopicPracticeFragmentTest.kt
@@ -256,24 +256,24 @@ class TopicPracticeFragmentTest {
         )
       ).perform(click())
       onView(
-      atPositionOnView(
-        recyclerViewId = R.id.topic_practice_skill_list,
-        position = 1,
-        targetViewId = R.id.subtopic_check_box
-      )
-    ).perform(click())
+        atPositionOnView(
+          recyclerViewId = R.id.topic_practice_skill_list,
+          position = 1,
+          targetViewId = R.id.subtopic_check_box
+        )
+      ).perform(click())
       onView(withId(R.id.topic_practice_skill_list)).perform(
         scrollToPosition<RecyclerView.ViewHolder>(
           5
         )
       )
       onView(
-      atPositionOnView(
-        recyclerViewId = R.id.topic_practice_skill_list,
-        position = 5,
-        targetViewId = R.id.topic_practice_start_button
-      )
-    ).perform(click())
+        atPositionOnView(
+          recyclerViewId = R.id.topic_practice_skill_list,
+          position = 5,
+          targetViewId = R.id.topic_practice_start_button
+        )
+      ).perform(click())
       testCoroutineDispatchers.runCurrent()
       intended(hasComponent(QuestionPlayerActivity::class.java.name))
       intended(hasExtra(QuestionPlayerActivity.getIntentKey(), skillIdList))

--- a/app/src/sharedTest/java/org/oppia/android/app/topic/practice/TopicPracticeFragmentTest.kt
+++ b/app/src/sharedTest/java/org/oppia/android/app/topic/practice/TopicPracticeFragmentTest.kt
@@ -245,48 +245,14 @@ class TopicPracticeFragmentTest {
 
   @Test
   fun testTopicPracticeFragment_loadFragment_selectSubtopics_clickStartButton_skillListTransferSuccessfully() { // ktlint-disable max-line-length
-//    testCoroutineDispatchers.unregisterIdlingResource()
+    testCoroutineDispatchers.unregisterIdlingResource()
     launchTopicActivityIntent(internalProfileId, FRACTIONS_TOPIC_ID).use {
-//      testCoroutineDispatchers.runCurrent()
-      onView(
-        allOf(
-          withText(TopicTab.getTabForPosition(position = 2, enablePracticeTab).name),
-          isDescendantOfA(withId(R.id.topic_tabs_container))
-        )
-      ).perform(click())
-      onView(
-        atPositionOnView(
-          recyclerViewId = R.id.topic_practice_skill_list,
-          position = 1,
-          targetViewId = R.id.subtopic_check_box
-        )
-      ).perform(click())
-      onView(withId(R.id.topic_practice_skill_list)).perform(
-        scrollToPosition<RecyclerView.ViewHolder>(
-          5
-        )
-      )
-      onView(
-        atPositionOnView(
-          recyclerViewId = R.id.topic_practice_skill_list,
-          position = 5,
-          targetViewId = R.id.topic_practice_start_button
-        )
-      ).perform(click())
-      testCoroutineDispatchers.runCurrent()
+      clickPracticeTab()
+      clickPracticeItem(position = 1, targetViewId = R.id.subtopic_check_box)
+      scrollToPosition(position = 5)
+      clickPracticeItem(position = 5, targetViewId = R.id.topic_practice_start_button)
       intended(hasComponent(QuestionPlayerActivity::class.java.name))
       intended(hasExtra(QuestionPlayerActivity.getIntentKey(), skillIdList))
-
-      /*clickPracticeTab()
-      testCoroutineDispatchers.runCurrent()
-      clickPracticeItem(position = 1, targetViewId = R.id.subtopic_check_box)
-      testCoroutineDispatchers.runCurrent()
-      scrollToPosition(position = 5)
-      testCoroutineDispatchers.runCurrent()
-      clickPracticeItem(position = 5, targetViewId = R.id.topic_practice_start_button)
-      testCoroutineDispatchers.runCurrent()
-      intended(hasComponent(QuestionPlayerActivity::class.java.name))
-      intended(hasExtra(QuestionPlayerActivity.getIntentKey(), skillIdList))*/
     }
   }
 

--- a/app/src/sharedTest/java/org/oppia/android/app/topic/practice/TopicPracticeFragmentTest.kt
+++ b/app/src/sharedTest/java/org/oppia/android/app/topic/practice/TopicPracticeFragmentTest.kt
@@ -92,7 +92,6 @@ import org.robolectric.annotation.Config
 import org.robolectric.annotation.LooperMode
 import javax.inject.Inject
 import javax.inject.Singleton
-import org.oppia.android.app.devoptions.DeveloperOptionsActivity
 
 /** Tests for [TopicPracticeFragment]. */
 @RunWith(AndroidJUnit4::class)

--- a/app/src/sharedTest/java/org/oppia/android/app/topic/practice/TopicPracticeFragmentTest.kt
+++ b/app/src/sharedTest/java/org/oppia/android/app/topic/practice/TopicPracticeFragmentTest.kt
@@ -244,8 +244,8 @@ class TopicPracticeFragmentTest {
   }
 
   @Test
-  @Ignore("Flaky test") // TODO(#3413): Test is failing unexpectedly.
   fun testTopicPracticeFragment_loadFragment_selectSubtopics_clickStartButton_skillListTransferSuccessfully() { // ktlint-disable max-line-length
+    testCoroutineDispatchers.unregisterIdlingResource()
     launchTopicActivityIntent(internalProfileId, FRACTIONS_TOPIC_ID)
     clickPracticeTab()
     clickPracticeItem(position = 1, targetViewId = R.id.subtopic_check_box)

--- a/app/src/sharedTest/java/org/oppia/android/app/topic/practice/TopicPracticeFragmentTest.kt
+++ b/app/src/sharedTest/java/org/oppia/android/app/topic/practice/TopicPracticeFragmentTest.kt
@@ -26,7 +26,6 @@ import org.hamcrest.Matchers.allOf
 import org.hamcrest.Matchers.not
 import org.junit.After
 import org.junit.Before
-import org.junit.Ignore
 import org.junit.Rule
 import org.junit.Test
 import org.junit.runner.RunWith

